### PR TITLE
 feat(plugin): add sandbox plugin

### PR DESCRIPTION
**Pull request by dmitriy-lukashevich**
## Description

Add plugin for safe execution of students tasks.
Firejail tool is used to make sandbox.

## Motivation and Context

Students code may try to make any changes which cause damage to the system. Also it can try to find such secret things as tokens or credentials which may be saved as environment variables.

This plugin intended to make runing of students scripts safety:
1. Hide all environment variables except of given.
2. Lock access to the Internet except of local sockets.
3. Prevent access to the file system except of givel folders and files.

Related to 
* #
* #
* 

Closes #

## Contribution Checklist

- [x] I have read and agreed on the [CONTRIBUTING](../CONTRIBUTING.md) and [CODE OF CONDUCT](../CODE_OF_CONDUCT.md) documents.
- [x] I have run linters, type checks, import sorting and tests locally.
- [x] I have added comments to code in hard-to-understand areas.
- [x] I have added 100% coverage tests for the new code (or aren't needed).
- [ ] I have added docs for the new code (or aren't needed).
- [ ] I have used [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) title: `tag(scope): description`.

### CODE REVIEW

File: checker/plugins/scripts.py [hunk 1]
```diff
@@ -17,20 +15,10 @@ class Args(PluginABC.Args):
         origin: str
         script: Union[str, list[str]]  # as pydantic does not support | in older python versions
         timeout: Union[float, None] = None  # as pydantic does not support | in older python versions
-        isolate: bool = False
-        env_whitelist: list[str] = Field(default_factory=lambda: ["PATH"])
```
Comment (dmitriy-lukashevich) [inline checker/plugins/scripts.py:21]: Кажется, теперь все элементы изолирования должны быть в отдельном плагине.

File: checker/plugins/firejail.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                   f'--whitelist={whitelist_dir}',
+                   # hide all environment variables
+                   'env -i',
+                   # copy PATH
+                   f'{PATH}={os.environ.get(PATH, "")}',
+                   # cope PYTHONPATH
+                   f'{PYTHONPATH}={os.environ.get(PYTHONPATH, "")}',
+                   args.script if isinstance(args.script, str) else ' '.join(args.script),
+                   ]
+        run_args = RunScriptPlugin.Args(origin=args.origin, script=' '.join(command), timeout=args.timeout)
```
Comment (k4black) [inline checker/plugins/firejail.py:37]: No need to join command, Script plugin (subprocess inside) accepts list

File: checker/plugins/firejail.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+PATH = 'PATH'
+PYTHONPATH = 'PYTHONPATH'
+
+
+class SafeRunScriptPlugin(RunScriptPlugin):
+    """Plugin for running scripts in safe mode."""
+
+    name = "safe_run_script"
+
+    Args = RunScriptPlugin.Args
```
Comment (k4black) [inline checker/plugins/firejail.py:17]: Seems it’s better to make custom Args - you need env whitelist, potentially network mode (eg for pip install profile or socket access) , custom allowed dirs -all with default parameters

File: checker/plugins/firejail.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                   # allow access to origin dir
+                   f'--whitelist={whitelist_dir}',
+                   # hide all environment variables
+                   'env -i',
+                   # copy PATH
+                   f'{PATH}={os.environ.get(PATH, "")}',
+                   # cope PYTHONPATH
+                   f'{PYTHONPATH}={os.environ.get(PYTHONPATH, "")}',
+                   args.script if isinstance(args.script, str) else ' '.join(args.script),
+                   ]
```
Comment (k4black) [inline checker/plugins/firejail.py:36]: Apply black for styling

File: checker/plugins/firejail.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                   # lock network access
+                   '--net=none',
+                   # allow access to origin dir
+                   f'--whitelist={whitelist_dir}',
+                   # hide all environment variables
+                   'env -i',
+                   # copy PATH
+                   f'{PATH}={os.environ.get(PATH, "")}',
+                   # cope PYTHONPATH
+                   f'{PYTHONPATH}={os.environ.get(PYTHONPATH, "")}',
```
Comment (k4black) [inline checker/plugins/firejail.py:34]: Potential untrusted code execution if spaces in the ENV. Seeems we need to secure with “” and or backslash all env content

File: checker/plugins/firejail.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+from .scripts import RunScriptPlugin
+
+
+PATH = 'PATH'
+PYTHONPATH = 'PYTHONPATH'
+
+
+class SafeRunScriptPlugin(RunScriptPlugin):
+    """Plugin for running scripts in safe mode."""
```
Comment (k4black) [inline checker/plugins/firejail.py:13]: Could you please elaborate a little explaining use of firejail

File: checker/plugins/firejail.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+    name = "safe_run_script"
+
+    Args = RunScriptPlugin.Args
+
+    # TODO(uniqum): at the moment "--queit" option of firejail may stil put extra strings in the output
+    def _run(self, args: Args, *, verbose: bool = False) -> str:
```
Comment (k4black) [inline checker/plugins/firejail.py:20]: Seems the problem we faced in tests can be tackled here - you need to check jirejail exists and through clear error msg

**Review by k4black — CHANGES_REQUESTED** [review]

**Comment by k4black** [comment]
пересоздай пожалуйста в main

**Pull request closed without merging**